checker/tests/run-pass/iterator.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+// A test that uses ranges mutated via iterators in a loop.
+
+#![feature(range_is_empty)]
+#![feature(unchecked_math)]
+
+#[macro_use]
+extern crate mirai_annotations;
+
+// pub fn test1() {
+//     let mut it = std::ops::Range {
+//         start: 0usize,
+//         end: 10usize,
+//     };
+
+//     while let Some(_) = it.next() {
+//         verify!(it.start <= 10);
+//     }
+//     verify!(it.start >= 10);
+// }
+
+// todo: sizes of expressions are too large. Simplify them on the fly.
```

Comment:
Perhaps worth checking if this is still needed.

Reply:
test1 and test2 are commented out because they somehow caused CI test failures. On my local machine, MIRAI can pass these tests, and the issue that expression sizes are too large still exists.